packages/pangea-sdk/pangea/services/vault/models/common.py

```diff
@@ -125,63 +149,63 @@ class CommonStoreResult(PangeaResponseResult):
     folder: Optional[str] = None
     metadata: Optional[Metadata] = None
     tags: Optional[Tags] = None
-    auto_rotate: Optional[bool] = None
-    rotation_policy: Optional[str] = None
-    store: Optional[bool] = None
+    rotation_frequency: Optional[str] = None
+    rotation_state: Optional[ItemVersionState] = None
     expiration: Optional[datetime.datetime] = None
-    managed: Optional[bool] = None
 
 
 class CommonGenerateResult(PangeaResponseResult):
     type: str
-    version: Optional[int] = None
-    id: Optional[str] = None
+    version: int
+    id: str
 
 
 class GetRequest(APIRequestModel):
     id: str
-    version: Optional[int] = None
+    version: Optional[Union[str, int]] = None
     verbose: Optional[bool] = None
+    version_state: Optional[ItemVersionState] = None
 
 
-class CommonGetResult(PangeaResponseResult):
-    type: str
-    id: str
+class ItemVersionData(PangeaResponseResult):
     version: int
-    name: Optional[str] = None
-    folder: Optional[str] = None
-    metadata: Optional[Metadata] = None
-    tags: Optional[Tags] = None
-    auto_rotate: Optional[bool] = None
-    rotation_policy: Optional[str] = None
-    last_rotated: Optional[str] = None  # TODO: should be time
-    next_rotation: Optional[str] = None  # TODO: should be time
-    expiration: Optional[str] = None  # TODO: should be time
-    created_at: Optional[str] = None  # TODO: should be time
-    revoked_at: Optional[str] = None  # TODO: should be time
-    retain_previous_version: Optional[bool] = None
+    state: str
+    created_at: str
+    destroy_at: Optional[str] = None
+    public_key: Optional[EncodedPublicKey] = None
+    secret: Optional[str] = None
 
 
-class ListItemData(APIRequestModel):
-    id: str
+class ItemData(PangeaResponseResult):
     type: str
-    managed: bool
-    last_rotated: Optional[str] = None  # TODO: should be time
-    next_rotation: Optional[str] = None  # TODO: should be time
-    expiration: Optional[str] = None  # TODO: should be time
-    rotation_policy: Optional[str] = None
-    identity: str
-    version: int
+    id: Optional[str] = None
+    item_state: Optional[str] = None
+    current_version: Optional[ItemVersionData] = None
     name: Optional[str] = None
     folder: Optional[str] = None
     metadata: Optional[Metadata] = None
     tags: Optional[Tags] = None
-    created_at: str  # TODO: should be time
-    revoked_at: Optional[str] = None  # TODO: should be time
+    rotation_frequency: Optional[str] = None
+    rotation_state: Optional[str] = None
+    last_rotated: Optional[str] = None
+    next_rotation: Optional[str] = None
+    expiration: Optional[str] = None
+    created_at: Optional[str] = None
+    algorithm: Optional[str] = None
+    purpose: Optional[str] = None
+
+
+class GetResult(ItemData):
+    versions: List[ItemVersionData] = []
+    rotation_grace_period: Optional[str] = None
+
+
+class ListItemData(ItemData):
+    compromised_versions: List[ItemVersionData]
```

Comment:
In /list endpoint when Type == "folder" you must not show compromised_versions field. This field should be Optional. Also add the corresponding logic to the endpoint.

		if respItem.Type != "folder" {
			respItem.ListResponseItemVerbose = &ListResponseItemVerbose{
				ID:                  item.ID,
				ItemState:           item.State,
				CreatedAt:           item.CreatedAt,
				Expiration:          item.Expiration,
				Tags:                item.Tags.Strings(),
				Metadata:            item.Metadata,
				RotationFrequency:   item.RotationFrequency,
				RotationState:       item.RotationState,
				LastRotated:         item.LastRotated,
				NextRotation:        item.NextRotation,
				Algorithm:           item.Algorithm,
				Purpose:             item.Purpose,
				CurrentVersion:      currentVersion[item.ID],
				CompromisedVersions: compromised,
			}
		}